resource: add `rv1exec_force` load format

jameshcorbett opened the pull request:
Problem: in some cases it can be useful to make the Fluxion resource module
initialize itself from rv1exec rather than JGF, even when the `scheduling`
key of R is present. Currently however, if the `scheduling` key is populated,
the resource module initializes itself with JGF.

Add a load-format, `rv1exec_force`, that tells the module to ignore the
`scheduling` key.

Checks the first box in #1414. There would need to be some configuration done in the system instance so that all its subinstances load `sched-fluxion-resource` with the `load-format=rv1exec_force` option. After chatting with @grondo, it sounds like:

> The best way to do this might be to have ansible drop an extension in to /etc/flux/modprobe/rc1.d/ that forces the option via:
> ```
> def setup(context):
>     context.setopt("sched-fluxion-resource", "OPT=VAL")
> ```

Using `context.attr_get("instance-level", 0) > 0` to check whether or not the script was running against the system instance.

jameshcorbett committed:
resource: add rv1exec_force load format

Problem: in some cases it can be useful to make the Fluxion resource module
initialize itself from rv1exec rather than JGF, even when the `scheduling`
key of R is present. Currently however, if the `scheduling` key is populated,
the resource module initializes itself with JGF.

Add a load-format, `rv1exec_force`, that tells the module to ignore the
`scheduling` key.

jameshcorbett committed:
t: add rq tests for `rv1exec_force` load format

Problem: there are no tests for the `rv1exec_force` load format.

Add basic sanity checks using resource-query.

milroy commented on t/t3039-resource-force-load-format.t:
```diff
@@ -0,0 +1,109 @@
+    head -n1 R1.out | jq -S "del(.execution.starttime, .execution.expiration)" > match1.json &&
+    jq -S "del(.execution.starttime, .execution.expiration)" R.json > R.norm.json &&
+    test_cmp match1.json R.norm.json
+'
```
The structure here is similar to what's done in the `3000`-series tests.

milroy reviewed (commented):
This PR is good overall, but the tests need some restructuring. I recommend extracting components of `t3039` that depend on `sched-fluion-resource` into a small `1000`-series test.

jameshcorbett committed:
t: add tests for `rv1exec_force` load format

Problem: there are no tests for the `rv1exec_force` load format.

Add basic sanity checks.

trws reviewed (approved):
Looks good to me.
